services/src/modules/directives/rest/datasource.ts

```diff
@@ -86,10 +89,16 @@ export class RESTDirectiveDataSource extends RESTDataSource<RequestContext> {
     args: GraphQLArguments,
     info: GraphQLResolveInfo
   ) {
-    if (!kvs || kvs.length == 0) return;
+    let validQueryParams = true;
+
+    if (!kvs || kvs.length == 0) return validQueryParams;
 
     for (const kv of kvs) {
       const value = inject(kv.value, parent, args, this.context, info);
+      if (!value && kv.required) {
```

Comment:
This doesn't work for `false`.
```suggestion
      if (value === undefined && kv.required) {
```

Reply:
value in the KeyValue type is a string, so it wouldn't let me set it as undefined or a boolean in the unit tests. When I changed this to value === undefined, it was still sending requests, and not entering into this failure statement.

Reply:
The `value` is result of argument evaluation. It can be anything. See the code 4 lines below.
https://github.com/Soluto/stitch/blob/fb8e62878d0ed4c89bc3ffe631aa965ce9a7d12e/services/src/modules/directives/rest/datasource.ts#L96

In `parseHeaders` function there is casting to `string`. https://github.com/Soluto/stitch/blob/fb8e62878d0ed4c89bc3ffe631aa965ce9a7d12e/services/src/modules/directives/rest/datasource.ts#L75
I think it can work for headers because despite the query params headers should be unique.